src/nix/path-info.cc

```diff
@@ -58,6 +64,25 @@ struct CmdPathInfo : StorePathsCommand, MixJSON
         };
     }
 
+    void printSize(int value)
+    {
+        if (!humanReadable) {
+            std::cout << '\t' << std::setw(11) << value;
+            return;
+        }
+
+        static constexpr std::array<char, 6> idents = {
+            ' ', 'K', 'M', 'G', 'T', 'P'
+        };
+        size_t power = 0;
+        double res = value;
+        while (res > 1024) {
+            ++power;
+            res /= 1024;
+        }
+        std::cout << '\t' << std::setw(11) << std::setprecision(3) << res << idents[power];
```

Comment:
Don't use iostream manipulators, they're stateful and ugly. Boost format strings are much nicer (e.g. `std::cout << fmt("%.3f", ...)` or something like that.

`idents[power]` looks like it could overflow in theory.

Reply:
Indeed, updated to be future proof in a > 1024 yottabyte world, and swapped to boost::format :+1: